admin/sections/config/chat.tsx

```diff
@@ -20,34 +20,41 @@ import { toast } from "@/components/ui/use-toast";
 import { zodResolver } from "@hookform/resolvers/zod";
 import { useEffect } from "react";
 import { useForm } from "react-hook-form";
-import { useQuery } from "react-query";
+import { useMutation, useQuery } from "react-query";
 
 export const ChatConfig = ({}: {}) => {
-  const form = useForm({
+  const form = useForm<ChatConfigFormType>({
     resolver: zodResolver(ChatConfigSchema),
   });
 
   // Use query to fetch the chat config
+  // TODO: show isLoading state, e.g. a spinner in the exapandable section component
   const { data, error, isLoading } = useQuery("chatConfig", getChatConfig, {
     refetchOnWindowFocus: false,
   });
+  const { mutate: updateConfig, updateError } = useMutation(updateChatConfig, {
+    onError: (error: unknown) => {
+      // TODO: form.reset(data) is just resetting to the last value, but a failure in the API
+      // call doesn't guarantee that the data is still the same.
+      // @Lee can we guarantee this? if not we should probably refetch the data
```

Comment:
Yeah, the chat config doesn't run any data tasks, so it's fine to keep the config as is, if there's an error.

Reply:
@thucpn then you can have a look at this task if you have time